scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -36,6 +36,8 @@ import scala.util.Failure
 class ScalametaParser(input: Input, dialect: Dialect) { parser =>
   require(Set("", EOL).contains(dialect.toplevelSeparator))
   implicit val currentDialect: Dialect = dialect
+  val soft = new SoftKeywords(currentDialect)
```

Comment:
somewhat optional: 1. `private` 2. use `dialect` directly, and not the intermediate `currentDialect`?

also, for a completely different, unrelated change: we could change the class to
```
class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   ...
```
and remove `currentDialect`.

Reply:
Good idea! Changed!